site/app/libraries/DiffViewer.php

```diff
@@ -150,10 +167,20 @@ public function buildViewer() {
                 $this->expected_file_image = $expected_file;
             }
             else{
-                $this->expected = file_get_contents($expected_file);
-                $this->has_expected = trim($this->expected) !== "" ? true : false;
-                $this->expected = explode("\n", $this->expected);
-                $this->display_expected = true;
+                 if(filesize($actual_file) < $size_limit){
+                    $this->expected = file_get_contents($expected_file);
+                    $this->has_expected = trim($this->expected) !== "" ? true : false;
+                    $this->expected = explode("\n", $this->expected);
+                    $this->display_expected = true;
+                }
+                else{
+                    $can_diff = false;
+                    //load in the first sizelimit characters of the file (TEMP VALUE)
+                    $this->expected = file_get_contents($expected_file, NULL, NULL, 0, $size_limit);
+                    $this->has_expected = trim($this->expected) !== "" ? true : false;
+                    $this->expected = explode("\n", $this->expected);
+                    $this->display_expected = true;
+                }
```

Comment:
what happens when you do `file_get_contents($expected_file, NULL, NULL, 0, $size_limit);` on a file that's size is less than `$size_limit`?

Reply:
Good catch. As it turns out, nothing interesting happens so far as I can tell. However, I should have been comparing the expected file's size to the size limit rather than the actual file's size, which stops us from ever getting into the state you mentioned. I've cleaned it up in a new pull request.